### DISCUSSION TITLE
how to copy the .py file into $PATH

### DISCUSSION THREAD

**Discussion by zuiop11** [newren/git-filter-repo#181] [Q&A] [answered]
hi. sorry if this is not trivial to me. i copied the 3000 lines of code into a python script and saved it next to my local repository that i want to filter. when i go into the local repo and use `git filter-repo --path <path-to-delete> --invert-path`, unsurprisingly git does not know this command. could you be more precise on that one.

**Comment by newren** [comment, marked as answer]
When you run "git CMD" and CMD is not something that git recognizes as an internal built in (e.g. commit, add, diff, log), then git looks for an executable or a script named "git-CMD".  git isn't going to search your entire harddrive for such a thing though, instead it only looks in pre-recorded places where you want the system to look for executables.  This list of locations is in a variable known as "PATH".

You can lookup more information about how PATH works at https://en.wikipedia.org/wiki/PATH_(variable).

In most shells on Linux/MacOSX/BSD/etc., you'd modify the path with a command that looks like the following:
   export PATH=$PATH:/directory/where/you/placed/the/new/script

You don't have to run git-filter-repo that way, though.  You can always just run the script directly instead of trying to get git to call it: `git-filter-repo --path <path-to-delete> --invert-paths`

**Comment by Gregor959** [comment]
On Windows I had to copy git-filter-repo.py and git-filter-repo into the git repository I wanted to work in and then
just run python.exe assuming python.exe is python3
python3.exe  git-filter-repo
e.g.
python.exe git-filter-repo --invert-paths --paths-from-file filesIDontWantAnymore.txt
